frontend/packages/knative-plugin/src/topology/create-connector-utils.ts

```diff
@@ -0,0 +1,49 @@
+import { Node } from '@patternfly/react-topology';
+import { MenuOptions } from '@console/dev-console/src/utils/add-resources-menu-utils';
+import { addResourceMenuWithoutCatalog } from '@console/dev-console/src/actions/add-resources';
+import { GraphData, getResource } from '@console/dev-console/src/components/topology';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { addEventSource } from '../actions/add-event-source';
+import { addTrigger } from '../actions/add-trigger';
+import { addSubscription } from '../actions/add-subscription';
+import { addPubSubConnectionModal } from '../components/pub-sub/PubSubModalLauncher';
+import { isEventingChannelResourceKind } from '../utils/fetch-dynamic-eventsources-utils';
+import {
+  ServiceModel,
+  EventingBrokerModel,
+  EventingSubscriptionModel,
+  EventingTriggerModel,
+} from '../models';
+
+export const getKnativeContextMenuAction = (
+  graphData: GraphData,
+  menu: MenuOptions,
+  connectorSource?: Node,
+): MenuOptions => {
+  if (isEventingChannelResourceKind(connectorSource?.getData().data.kind)) {
```

Comment:
nit : call `isEventingChannelResourceKind` only if `connectorSource` is defined

Reply:
Fixed all the comments, including filter section issue while the error message is set.